components/admin_console/team_channel_settings/team/list/team_list.jsx

```diff
@@ -28,136 +32,190 @@ export default class TeamList extends React.PureComponent {
-                        defaultMessage='Name'
-                    />
-                </div>
-                <div className='group-content'>
-                    <div className='group-description adjusted'>
-                        <FormattedMessage
-                            id='admin.team_settings.team_list.mappingHeader'
-                            defaultMessage='Management'
-                        />
-                    </div>
-                    <div className='group-actions'/>
-                </div>
-            </div>
-        );
+    componentDidMount() {
+        this.loadPage(0);
     }
 
-    searchBar = () => {
-        return (
-            <div className='groups-list--global-actions'>
-                <div className='group-list-search'>
-                    <input
-                        type='text'
-                        placeholder={Utils.localizeMessage('search_bar.search', 'Search')}
-                        onKeyUp={this.handleTeamSearchKeyUp}
-                        onChange={this.searchBarChangeHandler}
-                        value={this.state.searchString}
-                        data-testid='search-input'
-                    />
-                    <SearchIcon
-                        id='searchIcon'
-                        className='search__icon'
-                        aria-hidden='true'
-                    />
-                    <i
-                        className={'fa fa-times-circle group-filter-action ' + (this.state.searchString.length ? '' : 'hidden')}
-                        onClick={this.resetSearch}
-                        data-testid='clear-search'
-                    />
-                </div>
-            </div>
-        );
+    getPaginationProps = () => {
+        const {page, total} = this.state;
+        const startCount = (page * PAGE_SIZE) + 1;
+        let endCount = (page + 1) * PAGE_SIZE;
+        endCount = endCount > total ? total : endCount;
+        return {startCount, endCount, total};
     }
 
-    searchBarChangeHandler = (e) => {
-        const {searchString} = this.state;
-        if (searchString.length !== 0 && e.target.value.length === 0) {
-            this.resetSearch();
-            return;
+    loadPage = async (page, term = '') => {
+        this.setState({loading: true});
+        let response, teams, total;
+
+        if (term.length > 0) {
+            response = await this.props.actions.searchTeams(term, page, PAGE_SIZE);
+            teams = response.data.teams;
+            total = response.data.total_count;
+        } else {
+            response = await this.props.actions.getData(page, PAGE_SIZE);
+            teams = response.data.teams;
+            total = response.data.total_count;
         }
-        this.setState({searchString: e.target.value});
-    }
 
-    handleTeamSearchKeyUp = async (e) => {
-        const {key} = e;
-        const {searchString} = this.state;
+        this.setState({page, loading: false, teams, total, term});
```

Comment:
I think we should still continue using the redux state and pass teams / total / term as props